### PR TITLE
[CDAP-19350] Add source tar to distributed image

### PR DESCRIPTION
**Pull request by dli357**
[CDAP-19350] Add source tar to distributed image

[CDAP-19350] Add .git and node_modules to .gitignore for tar

[CDAP-19350] Tar specific subdirectories and files and use --exclude-vcs

[CDAP-19350]: https://cdap.atlassian.net/browse/CDAP-19350?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CDAP-19350]: https://cdap.atlassian.net/browse/CDAP-19350?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CDAP-19350]: https://cdap.atlassian.net/browse/CDAP-19350?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -19,22 +19,27 @@ ENV MAVEN_OPTS -Xmx2048m -Dhttp.keepAlive=false
...
+    -Dmetricswriters.extensions.dir="$DIR/metricswriters-extensions" \
+    -Deventwriters.extensions.dir="$DIR/eventwriters-extensions" \
     -Dui.build.name=cdap-non-optimized-full-build
 
 FROM openjdk:8-jdk AS run
 WORKDIR /
 COPY --from=build /cdap/build/cdap/cdap-master/target/stage-packaging/opt/cdap/master /opt/cdap/master
 COPY --from=build /cdap/build/cdap/cdap-ui/target/stage-packaging/opt/cdap/ui /opt/cdap/ui
 COPY --from=build /cdap/build/cdap/cdap-distributions/src/etc/cdap/conf.dist/logback*.xml /opt/cdap/master/conf/
+COPY --from=build /cdap/build/cdap/cdap-distributions/src/COPYRIGHT /opt/cdap/COPYRIGHT
```
Comment (albertshau) [inline Dockerfile:41]: a copyright is different than a license, let name it licenses?

**Review by albertshau — APPROVED** [review]

File: Dockerfile (hunk 1 above)
Comment (dli357) [reply Dockerfile]: Done

File: Dockerfile [hunk 2]
```diff
@@ -43,13 +48,21 @@ RUN apt-get update && \
...
   curl -L -o /opt/cdap/master/ext/jdbc/postgresql/postgresql-42.2.5.jar https://jdbc.postgresql.org/download/postgresql-42.2.5.jar && \
   curl -L -o /opt/cdap/master/ext/jdbc/postgresql/postgres-socket-factory-1.0.12-jar-with-dependencies.jar https://github.com/GoogleCloudPlatform/cloud-sql-jdbc-socket-factory/releases/download/v1.0.12/postgres-socket-factory-1.0.12-jar-with-dependencies.jar && \
   tar -xzf /opt/hadoop/hadoop-2.9.2.tar.gz -C /opt/hadoop && \
   curl -L -o /opt/hadoop/hadoop-2.9.2/share/hadoop/common/lib/hadoop-aws-2.9.2.jar https://repo1.maven.org/maven2/org/apache/hadoop/hadoop-aws/2.9.2/hadoop-aws-2.9.2.jar && \
   curl -L -o /opt/hadoop/hadoop-2.9.2/share/hadoop/common/lib/aws-java-sdk-bundle-1.11.199.jar https://repo1.maven.org/maven2/com/amazonaws/aws-java-sdk-bundle/1.11.199/aws-java-sdk-bundle-1.11.199.jar && \
   tar -xzf /opt/spark/spark-3.1.1-bin-without-hadoop.tgz -C /opt/spark && \
+  # Copy downloaded JAR licenses to COPYRIGHT folder
+  mkdir /opt/cdap/COPYRIGHT/hadoop-2.9.2 && cp /opt/hadoop/hadoop-2.9.2/*.txt /opt/cdap/COPYRIGHT/hadoop-2.9.2 && \
```
Comment (albertshau) [inline Dockerfile:59]: should rename these folders too for consistency

Comment (dli357) [reply Dockerfile]: Fixed

**Pull request merged**